### ISSUE TITLE
[java] ArrayIsStoredDirectly for enum constructor when array is not a compile-time constant

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#4870]
**Affects PMD Version:** 7.0.0

**Rule:** [ArrayIsStoredDirectly](https://docs.pmd-code.org/latest/pmd_rules_java_bestpractices.html#arrayisstoreddirectly)

**Description:**

The original array could be modified, even though we use it in an enum constant.

**Code Sample demonstrating the issue:**

```java
import java.util.Arrays;

public class ArrayIsStoredDirectlyTest {
    public void testEnum() {
        System.out.println(Test.A);
        System.out.println(Test.B);
        System.out.println(Test.C);

        strings[1] = "modified";
        System.out.println(Test.C);
    }

    public static String[] strings = new String[]{"a", "b"};

    public enum Test {
        A("a"),

        B("b".toUpperCase()),

        C(strings);

        private final String[] a;
        Test(String ... args) {
            a = args;  // <---- violation expected here
        }

        @Override
        public String toString() {
            return super.toString() + ":" + Arrays.toString(a);
        }
    }
}
```

**Expected outcome:**

PMD should report a violation at line ..., but doesn't. This is a false-negative.

**References:**

Related to #1413
